jupyterhub_config.py

```diff
@@ -46,6 +46,13 @@
         'profile_options': {
             'image': {
                 'display_name': 'Image',
+                'other_choice': {
```

Comment:
`other_choice` always seemed a bit *too* generic to me, as `other` feels super generic a term. I spent some more time thinking about this, and I think I'd prefer calling it `unlisted_choice`, as it makes it clearer there is a *list* of choices from the admin, and this enables the end user to make a choice that is *unlisted*, but still possibly supported with some validation.

Reply:
I'm not sure what I think is the best naming yet. I agree there is benefit of understanding that `profile_options` can involve a dropdown list of choices, but it seems a bit out of scope to let this name, which may or may not show up in a config, help convey that we have a list of choices.

I realize I've had a hidden assumption: that this new feature can be used without entries in the `choices` list. I think this is related to the naming discussion because it influences how coupled this choice is to the `choices` list.

The new config in this PR can be used when no entries are in `choices`, it just comes with some redundant UI:

### `choices` with _two_ entries, `other_choice` enabled

![current](https://github.com/jupyterhub/kubespawner/assets/3837114/eb3c9a2d-05b8-4c45-bebc-342d11cdc587)

### `choices` with _no_ entries, `other_choice` enabled

![current-with-no-listed-choices](https://github.com/jupyterhub/kubespawner/assets/3837114/739de440-82c0-455d-ad76-87f49b333150)

---

@yuvipanda and @batpad here are some decisions I think would be helpful to make before deciding on the naming:

1. If the new config `other_choice`/`unlisted_choice` be allowed when `choices` is empty
2. What UI to go for in those situations
3. If it should say "Other..." or something else by default in the dropdown list when using `other_choice`/`unlisted_choice`
4. If that "Other..." text should be configurable or not

I think (1) it would be great to support use of the new config when `choices` is empty, and (2) that it would be great to not show a single-value dropdown list in the case that the new config is enabled with `choices` empty. I'm not sure about (3) and (4).

Reply:
@consideRatio I agree with you on 1 and 2, and should probably be configurable as well (4). I think (Other) is a reasonable default when we do have a list of choices. However, I think none of these should block this PR getting merged, and can be done as follow up PRs.

I think this being usable without any listed choices and the ability to configure the “other” text makes the case stronger for calling it “unlisted_choice” than “other_choice”.

Reply:
If it makes any difference, I think `unlisted_choice` it's more accurate with respect to what it represents.
But I don't think strongly about naming, as long as we have docs that do a good job describing what it does.

Reply:
I'd love for us to go with `unlisted_choice`!

Reply:
I opened https://github.com/jupyterhub/kubespawner/issues/757 and https://github.com/jupyterhub/kubespawner/issues/756 to track these.